sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -115,12 +117,18 @@ def __init__(self, name, init_config, agentConfig, instances=None):
             'rate' : self.rate,
             'histogram': self.histogram
         }
+        self.adoprovider = self.default_adoprovider
 
         self.connector = init_config.get('connector', 'adodbapi')
         if not self.connector.lower() in self.valid_connectors:
             self.log.error("Invalid database connector %s, defaulting to adodbapi" % self.connector)
             self.connector = 'adodbapi'
 
+        self.adoprovider = init_config.get('adoprovider', self.default_adoprovider)
+        if not self.adoprovider.upper() in self.valid_adoproviders:
+            self.log.error("Invalid ADODB provider string %s, defaulting to %s" % (self.adoprovider, self.default_adoprovider))
```

Comment:
I think we should move over to `str.format()` and `{}`s.

Reply:
all the rest of the messages are this format; if we want to upgrade IMO that's a separate PR